std/typecons.d

```diff
@@ -3109,6 +3109,71 @@ struct Nullable(T)
     {
         return isNull ? fallback : _value.payload;
     }
+
+    // Range interface
+    /**
+     * Returns whether or not any values remain in this range.
+     */
+    alias empty = isNull;
+    /**
+     * Advances this range.
+     */
+    alias popFront = nullify;
+    /// ditto
+    alias popBack = nullify;
+    /**
+     * Returns the range's current value.
+     */
+    @property ref inout(T) front() inout @safe pure nothrow
+    {
+        return get();
+    }
+    /// ditto
+    alias back = front;
+    ///
+    @property inout(typeof(this)) save()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()(size_t from, size_t to) inout
+    {
+        import core.exception : RangeError;
+        if(from > length || to > length || to < from) {
+            throw new RangeError(__FILE__, __LINE__);
+        }
+        return (from == 0 && to == 1) ? this : this.init;
```

Comment:
```suggestion
        return (from != to) ? this : this.init;
```

Reply:
I can see how this is correct, but it took longer to figure that out...